api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/Eri4MonthsCohortQueries.java

```diff
@@ -58,62 +58,48 @@ public class Eri4MonthsCohortQueries {
-            + "       AND e.encounter_datetime BETWEEN Date_add(inicio_real.first_pickup, "
-            + "                                        INTERVAL 61 day) "
-            + "                                        AND "
-            + "                                            Date_add( "
-            + "                                            inicio_real.first_pickup, INTERVAL "
-            + "                                            120 day))"
-            + "       OR (e.encounter_type = ${mastercardDrugPickupEncounterType} "
-            + "       AND o.value_coded = ${artDatePickupMasterCard}  "
-            + "       AND o.value_datetime BETWEEN Date_add(inicio_real.first_pickup, "
-            + "                                        INTERVAL 61 day) "
-            + "                                        AND "
-            + "                                            Date_add( "
-            + "                                            inicio_real.first_pickup, INTERVAL "
-            + "                                            120 day)))"
-            + "GROUP  BY inicio_real.patient_id ";
+            + "        ) AS inicio_real "
+            + "WHERE  inicio_real.first_pickup BETWEEN :startDate AND :endDate "
+            + "  AND EXISTS (SELECT e.patient_id "
+            + "              FROM   encounter e "
+            + "                         INNER JOIN obs o "
+            + "                                    ON e.encounter_id = o.encounter_id "
+            + "              WHERE  e.patient_id = inicio_real.patient_id "
+            + "                AND e.voided = 0 "
+            + "                AND o.voided = 0 "
+            + "                AND e.location_id = :location "
+            + "                AND ( ( e.encounter_type = ${18} "
+            + "                  AND e.encounter_datetime BETWEEN "
+            + "                            inicio_real.first_pickup + INTERVAL 61 "
+            + "                                day "
+            + "                            AND "
+            + "                            inicio_real.first_pickup + INTERVAL "
+            + "                                120 "
+            + "                                day ) "
+            + "                  OR ( e.encounter_type = ${52} "
+            + "                      AND o.concept_id = ${23866} "
+            + "                      AND o.value_datetime BETWEEN "
+            + "                           inicio_real.first_pickup "
+            + "                               + "
+            + "                           INTERVAL 61 "
+            + "                               day "
+            + "                           AND "
+            + "                           inicio_real.first_pickup + INTERVAL "
+            + "                               120 "
+            + "                               day ) ))";
```

Comment:
Please check this interval statements, looks like the function is missing (DATE_ADD/DATE_SUB)